### DISCUSSION TITLE
SearchParams not loading for nested dynamic routes using Catch-all Segments"

### DISCUSSION THREAD

**Discussion by sandeepbfnp** [vercel/next.js#69656] [App Router]
In the given screenshot , i haved used the "Catch-all segments" while using dynamic routing but the searchParams is not loading for the src/app/(root)/(domestic)/(pdp)/gift/[...slugs].page.jsx
![image](https://github.com/user-attachments/assets/e5f79886-8fe6-434a-8bf9-225e6e524a83)
![Screenshot from 2024-09-04 13-44-42](https://github.com/user-attachments/assets/385095e2-a863-4b15-839c-204cea02f213)

**Comment by Juneezee** [comment]
> but the searchParams is not loading

What does your 

```js
console.log(searchParams, "searchParams")
```

prints?

**Reply by sandeepbfnp** [reply]
doesn't print on initial Load.
printing the value on the re-render of component where this console is printed
